CSharpSourceCode/HarmonyPatches/CampaignPatches.cs

```diff
@@ -55,21 +58,108 @@ public static bool Prefix(string id, MBObjectManager __instance)
+        public static bool SpawnWanderersAtStart(UrbanCharactersCampaignBehavior __instance, ref List<Hero> ____companions, List<CharacterObject> ____companionTemplates)
         {
-            if (companionTemplate == null || !companionTemplate.Culture.IsMainCulture)
+            List<CharacterObject> list = ____companionTemplates.Where(x => x.IsTOWTemplate()).ToList();
+            foreach (Settlement settlement in Settlement.All)
             {
-                return false;
+                if (settlement.IsTown)
+                {
+                    int targetNotableCountForSettlement = Campaign.Current.Models.NotableSpawnModel.GetTargetNotableCountForSettlement(settlement, Occupation.Artisan);
+                    for (int i = 0; i < targetNotableCountForSettlement; i++)
+                    {
+                        HeroCreator.CreateHeroAtOccupation(Occupation.Artisan, settlement);
+                    }
+                    int targetNotableCountForSettlement2 = Campaign.Current.Models.NotableSpawnModel.GetTargetNotableCountForSettlement(settlement, Occupation.Merchant);
+                    for (int j = 0; j < targetNotableCountForSettlement2; j++)
+                    {
+                        HeroCreator.CreateHeroAtOccupation(Occupation.Merchant, settlement);
+                    }
+                    int targetNotableCountForSettlement3 = Campaign.Current.Models.NotableSpawnModel.GetTargetNotableCountForSettlement(settlement, Occupation.GangLeader);
+                    for (int k = 0; k < targetNotableCountForSettlement3; k++)
+                    {
+                        HeroCreator.CreateHeroAtOccupation(Occupation.GangLeader, settlement);
+                    }
+                    for(int n = 0; n < list.Count; n++)
+                    {
+                        ____companions.Add(CreateTowWanderer(settlement, list[n]));
+                    }
+                }
+                else if (settlement.IsVillage)
+                {
+                    int targetNotableCountForSettlement4 = Campaign.Current.Models.NotableSpawnModel.GetTargetNotableCountForSettlement(settlement, Occupation.RuralNotable);
+                    for (int l = 0; l < targetNotableCountForSettlement4; l++)
+                    {
+                        HeroCreator.CreateHeroAtOccupation(Occupation.RuralNotable, settlement);
+                    }
+                    int targetNotableCountForSettlement5 = Campaign.Current.Models.NotableSpawnModel.GetTargetNotableCountForSettlement(settlement, Occupation.Headman);
+                    for (int m = 0; m < targetNotableCountForSettlement5; m++)
+                    {
+                        HeroCreator.CreateHeroAtOccupation(Occupation.Headman, settlement);
+                    }
+                }
             }
-            else return true;
+            return false;
+        }
+
+        private static Hero CreateTowWanderer(Settlement settlement, CharacterObject template)
```

Comment:
Is it necessary to have this private method here? Maybe move it to another class

Reply:
Its pretty much a copy-paste with adjustements of the TW original code and is tied directly with the harmonypatch. I would keep it here for quick reference. This is not used by anything else.